Setting non-integer value to a config bean attribute with @Min or @Max and without dataType=Integer.class locks the config bean.

Setting non-integer value to a config bean attribute with `@Min` or `@Max` and without `dataType=Integer.class` locks the config bean.
Then, the config bean attribute can't be changed until restart.

## Environment Details
* GlassFish Version: 4.1.2, 5.0-b08
* JDK version: 1.8.0_65
* OS: Windows 7

----------

## Problem Description
[//]: # "Describe the bug in detail highlighting current behavior vs expected behavior"
[//]: # "State if the problem is easily reproducible or happens intermittently"
[//]: # "Include stack traces or command outputs"
Setting steady-pool-size of jdbc-connection-pool to non-integer value fails with javax.validation.ValidationException: HV000028: Unexpected exception during isValid call..
```
> asadmin set resources.jdbc-connection-pool.DerbyPool.steady-pool-size=foo
remote failure: javax.validation.ValidationException: HV000028: Unexpected exception during isValid call.
HV000028: Unexpected exception during isValid call.
Command set failed.
```

In this case, these stack trace is logged to server.log.
```
[2017-06-14T13:24:18.661+0900] [glassfish 4.1] [SEVERE] [NCLS-CORE-00003] [javax.enterprise.system.core] [tid: _ThreadID=42 _ThreadName=admin-listener(1)] [timeMillis: 1497414258661] [levelValue: 1000] [[
  Exception while running a command
javax.validation.ValidationException: HV000028: Unexpected exception during isValid call.
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateSingleConstraint(ConstraintTree.java:286)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:133)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:91)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.validateConstraint(MetaConstraint.java:83)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraint(ValidatorImpl.java:547)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForDefaultGroup(ValidatorImpl.java:487)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForCurrentGroup(ValidatorImpl.java:451)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:403)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validate(ValidatorImpl.java:206)
	at org.jvnet.hk2.config.WriteableView.canCommit(WriteableView.java:316)
	at org.jvnet.hk2.config.Transaction.canCommit(Transaction.java:90)
	at org.jvnet.hk2.config.Transaction.commit(Transaction.java:109)
	at org.jvnet.hk2.config.ConfigSupport.apply(ConfigSupport.java:395)
	at com.sun.enterprise.v3.admin.SetCommand.set(SetCommand.java:481)
	at com.sun.enterprise.v3.admin.SetCommand.execute(SetCommand.java:178)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:539)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:535)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:356)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:534)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:557)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:356)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:556)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1464)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:109)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1846)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1722)
	at org.glassfish.admin.rest.resources.admin.CommandResource.executeCommand(CommandResource.java:404)
	at org.glassfish.admin.rest.resources.admin.CommandResource.execCommandSimpInMultOut(CommandResource.java:234)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:309)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:292)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1139)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:375)
	at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$3.service(JerseyContainerCommandService.java:173)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:179)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:206)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:180)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:283)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:200)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:132)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:111)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:536)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:591)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:571)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NumberFormatException: For input string: "foo"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:492)
	at java.lang.Integer.parseInt(Integer.java:527)
	at org.glassfish.jdbc.config.validators.JdbcConnectionPoolValidator.isValid(JdbcConnectionPoolValidator.java:83)
	at org.glassfish.jdbc.config.validators.JdbcConnectionPoolValidator.isValid(JdbcConnectionPoolValidator.java:59)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateSingleConstraint(ConstraintTree.java:283)
	... 73 more
]]
```

Then, setting jdbc-connection-pool.steady-pool-size to integer value (valid value) also fails until restarting the domain.
```
> asadmin set resources.jdbc-connection-pool.DerbyPool.steady-pool-size=8
remote failure: Could not change the attributes: Config bean already locked GlassFishConfigBean.org.glassfish.jdbc.config.JdbcConnectionPool
Config bean already locked GlassFishConfigBean.org.glassfish.jdbc.config.JdbcConnectionPool
Command set failed.
```

This problem occurrs when 
* the config bean attribute is annotated with `@Max` or `@Min`
* the config bean attribute is annotated with `@Attribute` but not specifying `dataType=Integer.class`
* setting non-integer value to the config bean attribute using asadmin set subcommand

For example, JdbcConnectionPool class has many config bean attributes that matches the above conditions.
```java
    @Attribute (defaultValue="8")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getSteadyPoolSize();

    @Attribute (defaultValue="32")
    @Min(value=1)
    @Max(value=Integer.MAX_VALUE)
    String getMaxPoolSize();

    @Attribute (defaultValue="60000")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getMaxWaitTimeInMillis();

    @Attribute (defaultValue="2")
    @Min(value=1)
    @Max(value=Integer.MAX_VALUE)
    String getPoolResizeQuantity();

    @Attribute (defaultValue="300")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getIdleTimeoutInSeconds();

    @Attribute (defaultValue="0")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getValidateAtmostOncePeriodInSeconds();

    @Attribute (defaultValue="0")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getConnectionLeakTimeoutInSeconds();

    @Attribute (defaultValue="0")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getConnectionCreationRetryAttempts();

    @Attribute (defaultValue="10")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getConnectionCreationRetryIntervalInSeconds();

    @Attribute (defaultValue="0")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getStatementCacheSize();

    @Attribute (defaultValue="0")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getStatementLeakTimeoutInSeconds();

    @Attribute (defaultValue="0")
    @Min(value=0)
    @Max(value=Integer.MAX_VALUE)
    String getMaxConnectionUsageCount();
```

Adding `dataType=Integer.class` to `@Attribute` fixes the issue.
But I think the root cause of this issue is the process of changing config bean attribute. (transaction, validation, locking and unlocking mechanism in HK2 or Hibernate Validator)

## Steps to reproduce
1. asadmin start-domain
1. asadmin set resources.jdbc-connection-pool.DerbyPool.steady-pool-size=foo
1. asadmin set resources.jdbc-connection-pool.DerbyPool.steady-pool-size=8

--- a/appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/JdbcConnectionPool.java
+++ b/appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/JdbcConnectionPool.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 1997, 2020 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Public License v. 2.0, which is available at
@@ -149,7 +150,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="8")
+    @Attribute(defaultValue = "8", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getSteadyPoolSize();
@@ -170,7 +171,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="32")
+    @Attribute(defaultValue = "32", dataType = Integer.class)
     @Min(value=1)
     @Max(value=Integer.MAX_VALUE)
     String getMaxPoolSize();
@@ -192,7 +193,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="60000")
+    @Attribute(defaultValue = "60000", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getMaxWaitTimeInMillis();
@@ -216,7 +217,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="2")
+    @Attribute(defaultValue = "2", dataType = Integer.class)
     @Min(value=1)
     @Max(value=Integer.MAX_VALUE)
     String getPoolResizeQuantity();
@@ -243,7 +244,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="300")
+    @Attribute(defaultValue = "300", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getIdleTimeoutInSeconds();
@@ -481,7 +482,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="0")
+    @Attribute(defaultValue = "0", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getValidateAtmostOncePeriodInSeconds();
@@ -509,7 +510,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="0")
+    @Attribute(defaultValue = "0", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getConnectionLeakTimeoutInSeconds();
@@ -551,7 +552,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="0")
+    @Attribute(defaultValue = "0", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getConnectionCreationRetryAttempts();
@@ -574,7 +575,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="10")
+    @Attribute(defaultValue = "10", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getConnectionCreationRetryIntervalInSeconds();
@@ -706,7 +707,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="0")
+    @Attribute(defaultValue = "0", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getStatementCacheSize();
@@ -745,7 +746,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="0")
+    @Attribute(defaultValue = "0", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getStatementLeakTimeoutInSeconds();
@@ -824,7 +825,7 @@ public interface JdbcConnectionPool extends ConfigBeanProxy, Resource, ResourceP
      * @return possible object is
      *         {@link String }
      */
-    @Attribute (defaultValue="0")
+    @Attribute(defaultValue = "0", dataType = Integer.class)
     @Min(value=0)
     @Max(value=Integer.MAX_VALUE)
     String getMaxConnectionUsageCount();

